client/extensions/woocommerce/state/ui/shipping/zones/methods/selectors.js

```diff
@@ -237,7 +239,21 @@ export const getCurrentlyOpenShippingZoneMethod = (
 		? false !== openMethod.enabled
 		: false !== zone.methods.currentlyEditingChanges.enabled;
 
-	return merge( {}, defaultValues, openMethod, zone.methods.currentlyEditingChanges, { enabled } );
+	// Overwrites the default behavior of `merge` while ignoring arrays and focusing on objects.
+	const customizer = ( objValue, srcValue ) => {
+		if ( isArray( objValue ) ) {
+			return srcValue;
+		}
+	};
+
+	return mergeWith(
+		{},
+		defaultValues,
+		openMethod,
+		zone.methods.currentlyEditingChanges,
+		{ enabled },
+		customizer
+	);
```

Comment:
It is necessary becuase if you have number-containing arrays, at least, sometimes the value gets doubled. Take a look at the example this way:

```js
const customizer = ( objValue, srcValue ) => {
	if ( _.isArray( objValue ) ) {
		return srcValue;
	}
};

const object = { a: [ 1, 4 ] };
const other = { a: [ 4 ] };

const withoutCustomizer = _.merge( object, other );
const withCustomizer = _.mergeWith( object, other, customizer );

withoutCustomizer.a
// [4, 4]

withCustomizer.a
// [4]
```

Lodash tries to merge both arrays and instead of leaving only `[4]` (as it should) or combining it into `[1,4]`, the result ends up being `[4,4]`. This interferes with the options for the shipping class, as when you try to delete a selected option, instead of going away it either stays or is even doubled (imagine trying to delete the option with ID `1` above.)

Reply:
I'm unable to reproduce the results from your example on a local Node REPL (in the `wp-calypso` repo), a Chrome console on the Lodash site, and in a Codepen.io. Could you perhaps replicate the behavior in a sharable code sandbox like Codepen or Runkit?

Reply:
I am not sure what random playground I used yesterday and cannot reproduce the issue with the snippet above anymore (super strange), but here is the CodePen: https://codepen.io/rageorgiev/pen/VBBzoY?editors=0011

Reply:
Ah, ok! I see the problem now. Let's add a comment illustrating the issue with merge preserving array values that are being removed and call it good. 👍

Reply:
Done 👍